libnetwork/etchosts/hosts.go

```diff
@@ -243,7 +243,11 @@ func parseExtraHosts(extraHosts []string) (HostEntries, error) {
 		if values[1] == "" {
 			return nil, fmt.Errorf("IP address in host entry %q is empty", entry)
 		}
-		e := HostEntry{IP: values[1], Names: []string{values[0]}}
+		ip := values[1]
+		if values[1] == "host-gateway" {
```

Comment:
Please split out host-gateway to a const at the top of the file and export it (starts with upper case)
Then you can just import than one in podman and we do not define the same string twice.

Reply:
I just recognized that HostContainersInternalIP is optional. Skipping the entry like the `host.containers.internal` one will be unexpected, and I should return an error. What do you think?

Reply:
returning an error sounds good